wallet/src/main/java/com/android/identity_credential/wallet/QrEngagementViewModel.kt

```diff
@@ -0,0 +1,99 @@
+package com.android.identity_credential.wallet
+
+import android.app.Application
+import androidx.core.content.ContextCompat
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.android.identity.android.mdoc.engagement.QrEngagementHelper
+import com.android.identity.android.mdoc.transport.DataTransport
+import com.android.identity.android.mdoc.transport.DataTransportOptions
+import com.android.identity.internal.Util
+import com.android.identity.mdoc.connectionmethod.ConnectionMethod
+import com.android.identity.mdoc.connectionmethod.ConnectionMethodBle
+import com.android.identity.securearea.SecureArea
+import com.android.identity.util.Logger
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.launch
+import java.security.KeyPair
+import java.util.UUID
+
+class QrEngagementViewModel(val context: Application) : AndroidViewModel(context)  {
+    companion object {
+        private const val TAG = "MainViewModel"
+    }
+
+    init {
+        Logger.i(TAG, "new mainViewModel")
+    }
+    var qrEngagementHelper: QrEngagementHelper? = null
+    private var eDeviceKeyPair: KeyPair? = null
+    private val eDeviceKeyCurve = SecureArea.EC_CURVE_P256
+
+    fun generateQrEngagementHelper(
+        onComplete: () -> Unit,
+        onConnected: () -> Unit)
+    {
+        Logger.i(TAG, "generateQrEngagementHelper called")
+        qrEngagementHelper?.close()
+        qrEngagementHelper = null
+
+        viewModelScope.launch(
+            CoroutineExceptionHandler{ _, exception ->
+                println("CoroutineExceptionHandler got $exception")
```

Comment:
Why `println()` and not `Logger.e()`? Also, should we be popping this error up to the UI or signal that it needs to navigate away from the QR screen or maybe even show the error to the user?

Reply:
updated, and now it will show "Error generating QR Code" to the user with the option to retry